### PR TITLE
feat: Document the SpokePool upgrade process

### PR DESCRIPTION
**Pull request by nicholaspai**
Explain high level how SpokePool upgrades will work

### CODE REVIEW

File: SpokePoolUpgrades.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+1. Deploy new SpokePool implementation contract on L2.
+2. Call `upgradeTo` on existing SpokePool proxy with new implementation address. The existing state should carry over
+   and the system should just work.
+
+## When upgrading without a proxy:
+
+If SpokePool is not an upgradeable proxy, then upgrading is more involved.
+
+1. Deploy new SpokePool contract.
```
Comment (mrice32) [inline SpokePoolUpgrades.md:19]: Should there be anything in here about dealing with assets that are in the spoke pool? Should they be transferred to the new spoke pool? Or back to the hub? You could send and execute a series of "relayer refunds" to transfer assets to the new spoke pool.

File: SpokePoolUpgrades.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# An Overview of how SpokePools would be upgraded
+
+## Before executing upgrade
+
+To be safe, make sure there is no pending bundle in the HubPool. To be extra safe, disable every single deposit route
+and only re-enable them after executing the upgrade. This is probably over-cautious when upgrading via proxy but
+definitely should be done when upgrading without a proxy.
```
Comment (mrice32) [inline SpokePoolUpgrades.md:7]: Are there any concerns about un-relayed deposits or outstanding partial fills?

**Review by mrice32 — APPROVED** [review]
This is awesome! I definitely think we should use this file as a place to build out the entire upgrade process. Great idea to commit it to the repo!

**Comment by nicholaspai** [comment]
Now that we just went through process we should document:

Upgrading/Adding totally new spoke pools
- Deploy spoke pools
- Transfer Ethereum_SpokePool ownership to HubPool
- Make sure all bundles are executed and there is none pending.
- Transfer all remaining spoke pool balance to new spoke pool
- `setCrossChainContracts`
- `setEnableDeposits=true` for every single route
- `whitelistTokens` for Arbitrum
- `setTokenBridge` for Optimism

Upgrading proxy spoke pools
- Run upgradeTo script to produce bytecode to pass through owner of HubPool
- Need to `relaySpokePoolAdmin` with data that contains `upgradeTo` instructions for each chain's proxy

**Pull request closed without merging**